In [1]:
import pandas as pd 
import numpy as np


In [22]:
# df=pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG (1).csv')
df=pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG (1).csv',encoding='latin-1')

In [14]:
df.head(10)

Area Code     Area  Item Code              Item  Element Code  \
0          4  Algeria       2501        Population           511   
1          4  Algeria       2501        Population          5301   
2          4  Algeria       2901       Grand Total           664   
3          4  Algeria       2901       Grand Total           674   
4          4  Algeria       2901       Grand Total           684   
5          4  Algeria       2903  Vegetal Products           664   
6          4  Algeria       2903  Vegetal Products           674   
7          4  Algeria       2903  Vegetal Products           684   
8          4  Algeria       2941   Animal Products           664   
9          4  Algeria       2941   Animal Products           674   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   
5           Food supply (kcal/capita/day)  kcal/capita/day   2932.00   
6  Protein supply quantity (g/capita/day)     g/capita/day     67.14   
7      Fat supply quantity (g/capita/day)     g/capita/day     53.84   
8           Food supply (kcal/capita/day)  kcal/capita/day    444.00   
9  Protein supply quantity (g/capita/day)     g/capita/day     27.76   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28  
5   2958.00   2941.00   2921.00   2932.00  
6     67.38     67.37     66.11     67.10  
7     54.72     52.04     55.59     54.57  
8    421.00    431.00    421.00    390.00  
9     26.97     27.35     26.71     24.73

In [41]:
df2 = df['Element'].unique()

QUES 1:- HERE WE CAN SEE THE REQUIRED DATA
Animal Fat produced in 2014-**209460.54**
IN 2017-**269617.53**

In [76]:
df2 = df.groupby('Item').sum()
df2

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                            
Alcohol, Non-Food        38447     752497       1515090    2403.00    2180.00   
Alcoholic Beverages      75016    1649136       2171381  102410.11   98783.72   
Animal Products          18060     397035         90990   11935.65   11811.73   
Animal fats              85279    1882494       2567303  209460.54  200675.72   
Apples and products      69280    1352989       1920418    9499.23   10559.15   
...                        ...        ...           ...        ...        ...   
Vegetables, Other        77779    1518715       2275409  155038.96  158104.08   
Vegetal Products         18060     391905         90990  107145.19  107064.17   
Wheat and products       87073    1639683       2654934  232670.13  234710.51   
Wine                     66206    1319535       1818328    4497.36    4251.81   
Yams                     50798     970905       1525745  200396.96  203151.78   

                         Y2016      Y2017      Y2018  
Item                                                  
Alcohol, Non-Food      2223.00    2348.00    2293.00  
Alcoholic Beverages   96958.75   95581.06   97847.27  
Animal Products       11661.69   11547.65   11578.61  
Animal fats          183314.22  269617.53  269648.27  
Apples and products    9853.26   10198.90    9640.51  
...                        ...        ...        ...  
Vegetables, Other    156218.90  157752.59  163987.21  
Vegetal Products     106792.72  107655.20  107775.39  
Wheat and products   239377.07  240047.62  242645.19  
Wine                   3872.09    4178.02    4039.32  
Yams                 220626.20  229174.59  221272.09  

[119 rows x 8 columns]

QUES-2
STD=1603.40
MEAN=135.23

In [78]:
df['Y2015'].std()

1603.4039839425013

In [79]:
df['Y2015'].mean()

135.23596649549626

QUES 3-
null=total_rows-not_null
nul=60943-59408
null=1535

In [81]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


QUES 4- HERE WE SEE THAT HIGHEST CRRELATIONI S WITH YEAR 2014

In [82]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

QUES-5-HERE WE CAN SEE HIGHEST IMPORT IS  OF  2017

In [83]:
df.groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

QUES-6,7-HERE WE CAN SEE PRODUCTION OF YEAR 2014 IS 1931287.75
FOR QUES 7- We can see that Domestic supply quantity has highest sum

In [84]:
df.groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

QUES-8-Here we can see that the 3rd lowest sum in 2018 is Protein supply quantity (g/capita/day)

In [87]:
df2=df.groupby('Element').sum()
df2.nsmallest(3,'Y2018')

Area Code  Item Code  Element Code  \
Element                                                                      
Tourist consumption                         50308    1486528       2869905   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   

                                           Y2014     Y2015     Y2016  \
Element                                                                
Tourist consumption                       416.00    349.00     89.00   
Fat supply quantity (g/capita/day)      10225.56  10235.74  10102.77   
Protein supply quantity (g/capita/day)  11836.46  11833.95  11779.69   

                                           Y2017     Y2018  
Element                                                     
Tourist consumption                        91.00     90.00  
Fat supply quantity (g/capita/day)      10253.84  10258.69  
Protein supply quantity (g/capita/day)  11842.45  11833.56

QUES-9- Here we can see that total Import Quantity in Algeria in 2018 is **36238.29**

In [88]:
df.groupby(['Area','Element']).sum().head(10)

Area Code  Item Code  \
Area    Element                                                     
Algeria Domestic supply quantity                   464     311204   
        Export Quantity                            392     264011   
        Fat supply quantity (g/capita/day)         432     291489   
        Feed                                        88      58967   
        Food                                       420     282744   
        Food supply (kcal/capita/day)              432     291489   
        Food supply quantity (kg/capita/yr)        420     282744   
        Import Quantity                            448     301030   
        Losses                                     208     139857   
        Other uses (non-food)                      148     101269   

                                             Element Code     Y2014     Y2015  \
Area    Element                                                                 
Algeria Domestic supply quantity                   614916  83755.94  85972.86   
        Export Quantity                            579278   1328.64   1104.62   
        Fat supply quantity (g/capita/day)          73872    320.19    317.40   
        Feed                                       121462   9656.04  10308.00   
        Food                                       539910  60582.69  61693.49   
        Food supply (kcal/capita/day)               71712  13500.00  13512.00   
        Food supply quantity (kg/capita/yr)         67725   1556.45   1552.81   
        Import Quantity                            628432  34940.12  37841.91   
        Losses                                     266396   7231.00   7471.00   
        Other uses (non-food)                      190698   3315.22   3483.37   

                                                Y2016     Y2017     Y2018  
Area    Element                                                            
Algeria Domestic supply quantity             87157.61  86810.42  90023.42  
        Export Quantity                       1340.22   1494.54    811.54  
        Fat supply quantity (g/capita/day)     309.54    320.74    309.09  
        Feed                                  9903.00   9571.00  10668.00  
        Food                                 63081.43  62785.33  63736.33  
        Food supply (kcal/capita/day)        13483.00  13363.00  13288.00  
        Food supply quantity (kg/capita/yr)   1555.77   1516.94   1509.28  
        Import Quantity                      36736.08  36059.29  36238.29  
        Losses                                7405.00   7428.00   7708.00  
        Other uses (non-food)                 3400.17   3534.08   3970.08

QUES-10-total number of unique countries in the dataset is 49

In [89]:
df2 = df['Area'].unique()
len(df2)

49